optimade/server/query_params.py

```diff
@@ -1,10 +1,52 @@
 from fastapi import Query
 from pydantic import EmailStr  # pylint: disable=no-name-in-module
-
+from typing import Iterable
 from optimade.server.config import CONFIG
-
-
-class EntryListingQueryParams:
+from warnings import warn
+from optimade.server.mappers import BaseResourceMapper
+from optimade.server.exceptions import BadRequest
+from optimade.server.warnings import UnknownProviderQueryParameter
+from abc import ABC
+
+
+class BaseQueryParams(ABC):
+    def check_params(self, query_params: Iterable[str]):
+        """This method checks whether all the query_parameters that are specified in the URL string occur in the relevant QueryParams class.
+        If a query parameter is found that is not defined in the QueryParams class and it does not have a known prefix, an appropriate error or warning will be given.
+
+        args:
+            query_params: An iterable object that returns the query parameters, as strings, for which it should be checked that they are in the relevant QueryParams class.
+
+        """
```

Comment:
Docstring tidying for the rendered docs

```suggestion
class BaseQueryParams(ABC):
    def check_params(self, query_params: Iterable[str]) -> None:
        """This method checks whether all the query parameters that are specified 
        in the URL string are implemented in the relevant `*QueryParams` class.
        
        If a query parameter is found that is not defined in the relevant `*QueryParams` class, 
        and it does not have a known provider prefix, an appropriate error (`BadRequest`) 
        or warning (`UnknownProviderQueryParameter`) will be emitted.

        Arguments:
            query_params: An iterable of the request's string query parameters.
            
        Raises:
            `BadRequest`: if the query parameter was not found in the relevant class, or if it
                does not have a valid prefix.

        """
```